### PR TITLE
[GPU] Adjust fusion count in DG2

### PR DESCRIPTION
**Pull request by tuxedcat**
### Details:
 - Activation won't be fused when onednn activation does not support it's activation_function since #12055

### Tickets:
 - 67491

### CODE REVIEW

File: src/plugins/intel_gpu/tests/fusions/gemm_fusion_test.cpp [hunk 1]
```diff
@@ -357,6 +360,9 @@ TEST_P(gemm_2in_act_scale_eltwise, broadcast_eltwise) {
         eltwise("sum", { "activation", "eltwise_data" }, eltwise_mode::sum,  data_types::f32),
         reorder("reorder_bfyx", "sum", p.default_format, data_types::f32)
     );
+    //Activation won't be fused because onednn doesn't support negative activation
+    if(engine.get_device_info().supports_immad)
+        p.expected_fused_primitives+=2;
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/tests/fusions/gemm_fusion_test.cpp:365]: Why +2? Won't activation + eltwise be fused?

Comment (tuxedcat) [reply src/plugins/intel_gpu/tests/fusions/gemm_fusion_test.cpp:365]: Yes. Current behavior is isolating activation node to make it select cldnn impl.

Comment (tuxedcat) [reply src/plugins/intel_gpu/tests/fusions/gemm_fusion_test.cpp:365]: https://github.com/openvinotoolkit/openvino/pull/12055#discussion_r917840956 This comment is outdated. Current implementation is option1 (do not fuse & do not be fused).

Comment (tuxedcat) [reply src/plugins/intel_gpu/tests/fusions/gemm_fusion_test.cpp:365]: No activation-eltwise fusion is intended behavior. That is +2 is correct. #11969 enabled it but it caused performance regression in some reason, and it was reverted. Refer fuse_eltwise_f in prepare_primitive_fusing.cpp.

File: src/plugins/intel_gpu/tests/fusions/activation_fusion_test.cpp [hunk 2]
```diff
@@ -174,6 +174,9 @@ TEST_P(activation_eltwise_activation_quantize_u8, basic) {
         quantize("quant", "act2", "in_low", "in_high", "out_low", "out_high", 256, data_types::u8),
         reorder("reorder_bfyx", "quant", p.default_format, data_types::f32)
     );
+    //Activation won't be fused because onednn doesn't support softsign activation
+    if(engine.get_device_info().supports_immad)
+        p.expected_fused_primitives++;
```
[the remark this answers is not in the document]
Comment (tuxedcat) [reply src/plugins/intel_gpu/tests/fusions/activation_fusion_test.cpp:179]: Done. I think lint does not run on test codes.

**Review by isanghao — APPROVED** [review]

**Pull request merged**